src/store/modules/storage.js

```diff
@@ -0,0 +1,50 @@
+import * as http from '../../plugins/http';
+import * as storage from '../../utils/storage';
+
+export const state = {
+  theme: 'blue',
+  darkMode: false,
+};
+
+export const mutations = {
+  changeTheme: (state, theme) => (state.theme = theme),
+  setDarkMode: (state, value) => (state.darkMode = value),
+  toggleDarkMode: state => (state.darkMode = !state.darkMode),
+};
+
+export const actions = {
+  init: async ({ commit }) => {
+    try {
+      // get and set config from local storage
+      const localTheme = storage.get('layout:theme');
+      const localDarkMode = storage.get('layout:dark-mode');
+      if (localTheme) commit('changeTheme', localTheme);
+      if (typeof darkMode === 'boolean') commit('setDarkMode', localDarkMode);
+      else if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) commit('setDarkMode', true);
+
+      // get and set config from ASF
+      // local config will be overwritten if ASF config is available
+      const response = await http.get('/storage/asfui-settings');
+      const { theme: asfTheme, darkMode: asfDarkmode } = response;
+      if (asfTheme) commit('changeTheme', asfTheme);
+      if (asfDarkmode) commit('setDarkMode', asfDarkmode);
```

Comment:
Now you are overriding local settings with remote

```suggestion
      // get and set config from ASF
      // local config will be overwritten if ASF config is available
      const response = await http.get('/storage/asfui-settings');
      const { theme: asfTheme, darkMode: asfDarkmode } = response;
      if (!localTheme && asfTheme) commit('changeTheme', asfTheme);
      if (!darkMode && asfDarkmode) commit('setDarkMode', asfDarkmode);
```

Reply:
Imagine we have 2 clients that connect to ASF-ui, we'll force them to use the same theme and dark mode settings, along with a flash of incorrect theme each time they visit. 

Local > Remote (previous):
- if 2nd client connects, it uses the same theme as 1st client (blue)
- if 2nd client changes theme (red), 1st client keeps old one (blue)
- if 3rd client connects, it uses the most recently changed theme (red)
- no matter how many times clients visit page, their preferred theme is kept, no action of a client affects other, already configured clients

Remote > Local (current):
- if 2nd client connects, it uses the same theme as 1st client (blue)
- if 2nd client changes theme (red), 1st client also changes theme (red) **!!**
- if 3rd client connects, it uses the most recently changed theme (red)
- with each reload, 1st client sees a flash of their preferred theme (blue) replaced by _global_ theme (red); in fact the worst case scenario all buy one client see flashes and wrong theme on each visit

Reply:
@JustArchi @MrBurrBurr @SuperSandro2000 Which approach do you think is better? Single ASF theme, or per-client configuration with initial state defaulting to latest change?

Reply:
I personally would go with single ASF theme.

Reply:
Considering the fact that ASF's IPC is meant for a single user exclusively, I'd go with single theme to not overcomplicate. Several people can already use several ASF instances, and there is very little motivation to design whole IPC permissions system to make it viable to have multiple-users configuration, moreover, it'd require far too many changes in core aspects of how ASF works, so I don't see it happening anytime soon, and probably never at all.